### PR TITLE
feature: Sync vector state to URL

### LINKED ISSUE #471: Sync vector state to URL
## Use Case

Spinoff from #469. Allow vector parameters to be saved to and read from the URL.

## Acceptance Criteria

_Please describe how you know this is done_

## Details

_Please provide any helpful specifications_

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #471, "sync vector state to URL."

*Estimated review size: small, 15 minutes*

Solution
========
- Adds handlers for serializing and deserializing vector URL state.
- Updates names for some decoding methods to simplify `url_utils` parsing.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-488/viewer?collection=https%3A%2F%2Fallencell.s3.amazonaws.com%2Faics%2Fnuc-morph-dataset%2Ftimelapse_feature_explorer_datasets%2Flineage-annotated_dataset%2Fcollection.json&dataset=Small&feature=volume&t=251&vc=1&vc-color=ff0000&vc-key=_motion_&vc-scale=18&vc-time-int=12&vc-tooltip=c
2. You should see the same settings as below: 

image

### CODE REVIEW

**Review by toloudis — APPROVED** [review]
LGTM!

File: src/colorizer/utils/url_utils.ts [hunk 1]
```diff
@@ -56,6 +59,12 @@ enum UrlParam {
   SCATTERPLOT_Y_AXIS = "scatter-y",
   SCATTERPLOT_RANGE_MODE = "scatter-range",
   OPEN_TAB = "tab",
+  SHOW_VECTOR = "vc",
+  VECTOR_KEY = "vc-key",
```
[the remark this answers is not in the document]
Comment (ShrimpCryptid) [reply src/colorizer/utils/url_utils.ts:63]: Yup, it's the equivalent!

**Comment by ascibisz** [comment]
When I was playing around on the preview link, almost everything was working as I would expect (just the basic changing something in the vector arrows menu, seeing the change in the UI, seeing that change in the URL, refreshing with that URL, seeing the state had been preserved in the UI because of the URL). 

However, when I un-check "Show vector arrows", I see the change in the UI, but I don't see the change in the URL, so of course when I refresh, the vector arrows are back again and the box is checked again. Is this expected? And is this happening for you too on the preview link?

**Comment by ShrimpCryptid** [comment]
> When I was playing around on the preview link, almost everything was working as I would expect (just the basic changing something in the vector arrows menu, seeing the change in the UI, seeing that change in the URL, refreshing with that URL, seeing the state had been preserved in the UI because of the URL).
> 
> However, when I un-check "Show vector arrows", I see the change in the UI, but I don't see the change in the URL, so of course when I refresh, the vector arrows are back again and the box is checked again. Is this expected? And is this happening for you too on the preview link?

Oh interesting, I'm seeing that behavior too. Let me debug this and I'll re-request you.

**Review by ascibisz — APPROVED** [review]

**Pull request merged**